### ISSUE TITLE
Can't handle parallel installs

### ISSUE THREAD

**Issue by ricab** [canonical/multipass#1129]
```console
$ snap install --classic --edge multipass_tmp
$ multipass_tmp version
/snap/multipass_tmp/1233/command-multipass.wrapper: 2: exec: /snap/multipass/1233/bin/launch-multipass: not found
```

**Comment by ricab** [comment]
Hi @Aryansingh0103, thanks for you interest! 

It's unfortunate that the issue description is pretty bare, but this was about making parallel snap installs work with Multipass. The issue is quite old so I am not sure what the current status is, nor how relevant the description above is anymore.  I guess the first thing would be to figure that out. 

Looking forward to your contribution here.

**Comment by Sploder12** [comment]
Parallel snap installing itself works, but Multipass does not in this configuration. (tested with 1.16.1)

**Comment by ricab** [comment]
There is probably a clash with `dnsmasq`, right? Any other obstacle you noticed?

**Comment by Sploder12** [comment]
yep, I did not look much further than that but things like certificate sharing might be an issue as well

**Comment by ricab** [comment]
OK. I think this is still relevant, though with low priority. It would be helpful for development and testing, so if we come to resolve `dnsmasq` port contention, it may become a low ripe fruit. 

Related to https://github.com/canonical/multipass/issues/2841
